src/ts/components/subtitleoverlay.ts

```diff
@@ -130,6 +145,49 @@ export class SubtitleOverlay extends Container<ContainerConfig> {
     subtitleClearHandler();
   }
 
+  detectCroppedSubtitleLabel(
+    labelElement: HTMLElement,
+  ): SubtitleCropDetectionResult {
+    const parent = this.getDomElement().get(0);
+
+    const childRect = labelElement.getBoundingClientRect();
+    const parentRect = parent.getBoundingClientRect();
+
+    return {
+      top: childRect.top < parentRect.top,
+      right: childRect.right > parentRect.right,
+      bottom: childRect.bottom > parentRect.bottom,
+      left: childRect.left < parentRect.left,
+    };
+  }
+
+  handleSubtitleCropping(label: SubtitleLabel) {
+    const labelDomElement = label.getDomElement();
+    const cropDetection = this.detectCroppedSubtitleLabel(
+      labelDomElement.get(0),
+    );
+
+    if (cropDetection.top) {
+      labelDomElement.css('top', '0');
+      labelDomElement.css('bottom', 'unset');
```

Comment:
Instead of setting this explicitly to `unset` - could we just remove it from the inline styles? Otherwise I'm not sure if external styling would still work, as the inline `unset` would override everything.

I guess we'd need another function in our DOM.ts but it should be easy enough.

Reply:
Good call! Implemented it as suggested:

[Remove inline css instead of overriding for caption overflows](https://github.com/bitmovin/bitmovin-player-ui/pull/538/commits/bc723697fed4937db120a334b86175ffe65c3f39)